Core.ApplicationServices/System/ItSystemService.cs

```diff
@@ -112,22 +123,66 @@ private IEnumerable<ItSystem> GetHierarchyParents(ItSystem itSystem)
             return parents;
         }
 
-        public void Delete(int id)
+        public SystemDeleteResult Delete(int id)
         {
-            // http://stackoverflow.com/questions/15226312/entityframewok-how-to-configure-cascade-delete-to-nullify-foreign-keys
-            // when children are loaded into memory the foreign key is correctly set to null on children when deleted
-            var system = _repository.Get(x => x.Id == id, null, $"{nameof(ItSystem.TaskRefs)}").FirstOrDefault();
+            var system = _itSystemRepository.GetSystem(id);
 
-            // delete it project
-            _repository.Delete(system);
-            _repository.Save();
-        }
+            if (system == null)
+            {
+                return SystemDeleteResult.NotFound;
+            }
 
-        public IEnumerable<ItSystem> ReportItSystemRights()
-        {
-            var rights = _rightsRepository.Get();
+            if (! _authorizationContext.AllowDelete(system))
+            {
+                return SystemDeleteResult.Forbidden;
+            }
+
+            if (system.Usages.Any())
+            {
+                return SystemDeleteResult.InUse;
+            }
 
-            return null;
+            if (system.Children.Any())
+            {
+                return SystemDeleteResult.HasChildren;
+            }
+
+            if (system.ItInterfaceExhibits.Any())
+            {
+                return SystemDeleteResult.HasInterfaceExhibits;
+            }
+
+            using (var transaction = _transactionManager.Begin(IsolationLevel.Serializable))
+            {
+                try
+                {
+                    var deleteReferenceResult = _referenceService.Delete(system.Id);
+                    switch (deleteReferenceResult)
```

Comment:
På det her tidspunkt er du kun interesseret i een ting.. Gik det godt? Hvis ikke -> unknown error. Den konkrete fejl fra reference service kan du logge sammen med Id på det system der blev forsøgt slettet, men dine individuelle håndteringer her går for langt